### PR TITLE
Small fix to correct the ownership of the synced_folder directory, re…

### PR DESCRIPTION
**Pull request by desrod**
This is a minor fix to avoid the 'Permission denied' message when the synced_folder directory is not owned by the '_apt' user and the downloaded files are not sandboxed.

### CODE REVIEW

File: virtual/Vagrantfile [hunk 1]
```diff
@@ -24,7 +24,7 @@ def mount_apt_cache(config)
   user_data_path = Vagrant.user_data_path.to_s
   cache_dir = File.join(user_data_path, 'cache', 'apt', config.vm.box)
   apt_cache_dir = '/var/cache/apt/archives'
-  config.vm.synced_folder cache_dir, apt_cache_dir, create: true
+  config.vm.synced_folder cache_dir, apt_cache_dir, create: true, owner: '_apt'
```
Comment (justinjpacheco) [inline virtual/Vagrantfile]: i don't think _apt will exist on osx systems

**Comment by justinjpacheco** [comment]
im not sure i follow here. where are you seeing this permission denied error?

File: virtual/Vagrantfile (hunk 1 above)
Comment (desrod) [reply virtual/Vagrantfile]: You'll see the following error: 

> r1n0: Download is performed unsandboxed as root as file '/var/cache/apt/archives/partial/iucode-tool_2.3.1-1_amd64.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)

The _apt user is known local to the VM itself, not the host. On macOS, _apt would be the user that vagrant maps the folder's ownership to inside the VM. apt downloads files within the context of the _apt user in a sandbox. Without that ownership, it downloads unsandboxed, as root. 

You won't see this if you've already populated your ~/.vagrant.d/cache/apt/bento/ubuntu-18.04 directory from previous invocations. Remove your cache directory completely and re-run a clean build. 

More context here: 
https://github.com/Debian/apt/blob/master/apt-pkg/acquire.cc#L656

**Comment by justinjpacheco** [comment]
can you provide some explicit instructions on how i can see/reproduce this error? i've never seen this before

**Comment by desrod** [comment]
1. On your build host: rm -rf ~/.vagrant.d/cache/
2. cd <your_bcpc_repo_root>/virtual
3. vagrant up

But I just realized this only appears when deploying using the 5.x HWE kernel (similar to what's been deployed on metal internally), and that kernel declares the iucode-tool package as a dependency. 

The reason I had to test a kernel upgrade was to try to resolve an issue with the e1000 driver in the default kernel in the current Bento box. There is a bug in that driver where it will hard panic the VM in an unrecoverable way. (see below)

The kernel in the current Bento is 4.15.0-29, although there are some 4.15.0-43 packages installed, the kernel matching those packages is not installed. Installing the 4.15.0-43 package did appear to resolve the panics with the e1000 driver in several iterations without fault. 

However, the 5.x HWE kernel is more robust and matches a portion of the deployed baremetal environment, but installing linux-image-generic should always track the most-stable 4.15 kernel. This kernel will continue to be supported until Bionic goes EOL in 4/2023, while the 5.x HWE kernel will sunset support when 19.10 goes EOL later in the year. 

Both of these kernels will install the iucode-tool package as a dependency, so you'll eventually run into this if/when you ever upgrade the kernel to something other than .29. 

These changes are covered in a feature branch I'll push shortly, where this same fix is included. I'll delete _this_ branch and continue in the other branch with this fix.

```
Jan 30 16:20:30 tor1 kernel: [   23.008430] ------------[ cut here ]------------
Jan 30 16:20:30 tor1 kernel: [   23.008441] NETDEV WATCHDOG: eth2 (e1000): transmit queue 0 timed out
Jan 30 16:20:30 tor1 kernel: [   23.008455] WARNING: CPU: 0 PID: 0 at /build/linux-60XibS/linux-4.15.0/net/sched/sch_generic.c:323 dev_watchdog+0x221/0x230
Jan 30 16:20:30 tor1 kernel: [   23.008457] Modules linked in: vboxsf(OE) intel_powerclamp intel_rapl_perf input_leds serio_raw mac_hid vboxguest(OE) ip6table_filter ip6_tables iptable_filter sch_fq_codel sunrpc ib_iser rdma_cm iw_cm ib_cm ib_core iscsi_tcp libiscsi_tcp libiscsi scsi_transport_iscsi ip_tables x_tables autofs4 btrfs zstd_compress raid10 raid456 async_raid6_recov async_memcpy async_pq async_xor async_tx xor raid6_pq libcrc32c raid1 raid0 multipath linear crct10dif_pclmul crc32_pclmul ghash_clmulni_intel pcbc aesni_intel aes_x86_64 crypto_simd glue_helper cryptd vboxvideo(OE) ttm drm_kms_helper syscopyarea sysfillrect sysimgblt fb_sys_fops psmouse ahci libahci i2c_piix4 drm e1000 pata_acpi video
Jan 30 16:20:30 tor1 kernel: [   23.008483] CPU: 0 PID: 0 Comm: swapper/0 Tainted: G           OE    4.15.0-29-generic #31-Ubuntu
Jan 30 16:20:30 tor1 kernel: [   23.008484] Hardware name: innotek GmbH VirtualBox/VirtualBox, BIOS VirtualBox 12/01/2006
Jan 30 16:20:30 tor1 kernel: [   23.008485] RIP: 0010:dev_watchdog+0x221/0x230
Jan 30 16:20:30 tor1 kernel: [   23.008485] RSP: 0018:ffff8d07dfc03e58 EFLAGS: 00010286
Jan 30 16:20:30 tor1 kernel: [   23.008486] RAX: 0000000000000000 RBX: 0000000000000000 RCX: 000000000000083f
Jan 30 16:20:30 tor1 kernel: [   23.008487] RDX: 0000000000000000 RSI: 00000000000000f6 RDI: 000000000000083f
Jan 30 16:20:30 tor1 kernel: [   23.008487] RBP: ffff8d07dfc03e88 R08: 0000000000000001 R09: 0000000000000210
Jan 30 16:20:30 tor1 kernel: [   23.008488] R10: ffff8d07dfc03ee0 R11: 0000000000000000 R12: 0000000000000001
Jan 30 16:20:30 tor1 kernel: [   23.008489] R13: ffff8d07ddeb2000 R14: ffff8d07ddeb2478 R15: ffff8d07de638280
Jan 30 16:20:30 tor1 kernel: [   23.008490] FS:  0000000000000000(0000) GS:ffff8d07dfc00000(0000) knlGS:0000000000000000
Jan 30 16:20:30 tor1 kernel: [   23.008490] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
Jan 30 16:20:30 tor1 kernel: [   23.008491] CR2: 00007fb2f1398c40 CR3: 000000001520a004 CR4: 00000000000606f0
Jan 30 16:20:30 tor1 kernel: [   23.008493] Call Trace:
Jan 30 16:20:30 tor1 kernel: [   23.008494]  <IRQ>
Jan 30 16:20:30 tor1 kernel: [   23.008496]  ? dev_deactivate_queue.constprop.33+0x60/0x60
Jan 30 16:20:30 tor1 kernel: [   23.008498]  call_timer_fn+0x30/0x130
Jan 30 16:20:30 tor1 kernel: [   23.008500]  run_timer_softirq+0x3fb/0x450
Jan 30 16:20:30 tor1 kernel: [   23.008502]  ? kvm_clock_get_cycles+0x1e/0x20
Jan 30 16:20:30 tor1 kernel: [   23.008503]  ? ktime_get+0x43/0xa0
Jan 30 16:20:30 tor1 kernel: [   23.008504]  ? native_apic_msr_write+0x2b/0x40
Jan 30 16:20:30 tor1 kernel: [   23.008506]  ? lapic_next_event+0x20/0x30
Jan 30 16:20:30 tor1 kernel: [   23.008527]  __do_softirq+0xdf/0x2b2
Jan 30 16:20:30 tor1 kernel: [   23.008529]  irq_exit+0xb6/0xc0
Jan 30 16:20:30 tor1 kernel: [   23.008530]  smp_apic_timer_interrupt+0x71/0x130
Jan 30 16:20:30 tor1 kernel: [   23.008531]  apic_timer_interrupt+0x84/0x90
Jan 30 16:20:30 tor1 kernel: [   23.008532]  </IRQ>
Jan 30 16:20:30 tor1 kernel: [   23.008533] RIP: 0010:mwait_idle+0x96/0x180
Jan 30 16:20:30 tor1 kernel: [   23.008533] RSP: 0018:ffffffffb9603e38 EFLAGS: 00000246 ORIG_RAX: ffffffffffffff11
Jan 30 16:20:30 tor1 kernel: [   23.008534] RAX: 0000000000000000 RBX: 0000000000000000 RCX: 0000000000000000
Jan 30 16:20:30 tor1 kernel: [   23.008535] RDX: 0000000000000000 RSI: 0000000000000000 RDI: 0000000000000000
Jan 30 16:20:30 tor1 kernel: [   23.008535] RBP: ffffffffb9603e50 R08: 00000007cdb4a865 R09: ffff8d07ddcedc00
Jan 30 16:20:30 tor1 kernel: [   23.008536] R10: 0000000000000000 R11: 000000d29ab390fb R12: 0000000000000000
Jan 30 16:20:30 tor1 kernel: [   23.008536] R13: 0000000000000000 R14: 0000000000000000 R15: 0000000000000000
Jan 30 16:20:30 tor1 kernel: [   23.008539]  arch_cpu_idle+0x15/0x20
Jan 30 16:20:30 tor1 kernel: [   23.008540]  default_idle_call+0x23/0x30
Jan 30 16:20:30 tor1 kernel: [   23.008542]  do_idle+0x172/0x1f0
Jan 30 16:20:30 tor1 kernel: [   23.008543]  cpu_startup_entry+0x73/0x80
Jan 30 16:20:30 tor1 kernel: [   23.008544]  rest_init+0xae/0xb0
Jan 30 16:20:30 tor1 kernel: [   23.008546]  start_kernel+0x4dc/0x4fd
Jan 30 16:20:30 tor1 kernel: [   23.008562]  x86_64_start_reservations+0x24/0x26
Jan 30 16:20:30 tor1 kernel: [   23.008563]  x86_64_start_kernel+0x74/0x77
Jan 30 16:20:30 tor1 kernel: [   23.008564]  secondary_startup_64+0xa5/0xb0
Jan 30 16:20:30 tor1 kernel: [   23.008565] Code: 38 00 49 63 4e e8 eb 92 4c 89 ef c6 05 27 36 d9 00 01 e8 03 38 fd ff 89 d9 48 89 c2 4c 89 ee 48 c7 c7 a8 76 39 b9 e8 df f3 80 ff <0f> 0b eb c0 90 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 44 00 00 55 
Jan 30 16:20:30 tor1 kernel: [   23.008583] ---[ end trace d6c1afb42a3259c1 ]---   
```

**Pull request merged**